src/main/java/org/example/afternoon_practice/PersonCSVExport.java

```diff
@@ -0,0 +1,133 @@
+            fileWriter.append('\n');
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void exportPersons(List<Person> users) {
+        List<String> usersStr = users.stream().map(Person::toString2).toList();
+        exportUsers(usersStr, false);
+    }
+
+    public static List<String> getUsersFromCSV() {
+        List<String> users = new ArrayList<String>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(USERS_FILE));){
+            String line = reader.readLine();
+            while (line != null) {
+                users.add(line);
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        return users;
+    }
+
+    public static void printUsers(List<String> users){
+        for (String user:users) {
+            System.out.println(user);
+        }
+    }
+
+    public static void getUsersNumberInCSV(){
+        int usersNumber = getUsersFromCSV().size();
+        System.out.println("CSV File contains " + usersNumber+ " users.");
+    }
+
+    public static void exportUsers(List<String> users, Boolean isAppendAllowed) {
+        try (FileWriter fileWriter = new FileWriter(USERS_FILE, isAppendAllowed);){
+            for (String user : users) {
+                fileWriter.append(user);
+                fileWriter.append('\n');
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void removeUser(String query, int columnIndex) {
+        List<String> users = getUsersFromCSV();
+        for (String user : users) {
+            String[] userArr = user.split(";");
+            if (userArr[columnIndex].equalsIgnoreCase(query)) {
+                users.remove(user);
+                break;
+            }
+        }
+        exportUsers(users, false);
+    }
+
+    public static void updateUser(String userName, String newData, int columnIndexToUpdate) {
+        List<String> users = getUsersFromCSV();
+        for (String user : users) {
+            String[] userArr = user.split(";");
+            if (userArr[0].equalsIgnoreCase(userName)) {
+                String updatedUser = user.replaceAll(userArr[columnIndexToUpdate], newData);
+                users.set(users.indexOf(user), updatedUser);
+                System.out.println("User was updated");
+                System.out.println(updatedUser);
+                break;
+            }
+        }
+        exportUsers(users, false);
+    }
+
+    public static List<String> searchUser(String query, int columnIndex) {
+        List<String> users = getUsersFromCSV();
+        return users.stream()
+                .map(user -> user.split(";"))
+                .filter(userArray -> userArray[columnIndex].equalsIgnoreCase(query))
+                .map(userArray -> String.join(";", userArray))
+                .toList();
```

Comment:
it's correct and a good solution but you still haven't touched streams AFAIK so this could be potentially done a bit simpler:
```suggestion
        for(String user: users){
              if(userArray[columnIndex].toLowerCase().contains(query))
                    return user; // assuming we only find one user with this query
         }
```